shashi/feb10.js

```diff
@@ -0,0 +1,37 @@
+// Write a function that creates an object with each (key, value) pair being the (lower case, upper case) versions of a letter, respectively.
+// Examples
+// mapping(["p", "s"]) ➞ { "p":"P", "s":"S" }
+// mapping(["a", "b", "c"]) ➞ { "a": "A", "b": "B", "c": "C" }
+// mapping(["a", "v", "y", "z"]) ➞ { "a": "A", "v": "V", "y": "Y", "z": "Z"}
+//Assumption: If any value of input array is not string, same value is added to key and value
+
+const mapping = arr => {
+  var obj = {};
+  
+  arr.forEach(element => {
+    if(typeof element === "string"){
+      //console.log("element in if", element)
+    obj[element.toLowerCase()]= element.toUpperCase();
+    } else {
+      //console.log("element in else", element)
+      obj[element]= element;
+    }
+  });
+  return JSON.stringify(obj);
```

Comment:
hmm why are you stringifying it?

Reply:
actually before that.. do you know what JSON.stringify() does?

Reply:
JSON.stringify converts an object to JSON format which is the expected format for this function. Otherwise the format is {a:”A”} and expected is {“a”:”A”}

Reply:
The key of JSON.stringify  is it actually converts everything into a string. The requirement is to convert into  an object. So the output should be an object. In an object, keys are represented as a string.